components/mailgun/actions/send-email/send-email.js

```diff
@@ -0,0 +1,104 @@
+const mailgun = require("../../mailgun.app.js");
+
+module.exports = {
+  key: "mailgun-send-email",
+  name: "Mailgun Send Email",
+  description: "Send email with Mailgun.",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    mailgun,
+    domain: {
+      propDefinition: [
+        mailgun,
+        "domain",
+      ],
+    },
+    fromName: {
+      type: "string",
+      label: "From Name",
+      description: "Sender name",
+    },
+    from: {
+      type: "string",
+      label: "From Email",
+      description: "Sender email address",
+    },
+    replyTo: {
+      type: "string",
+      label: "Reply Email",
+      description: "Sender Reply-To email address",
+    },
+    to: {
+      type: "string[]",
+      label: "To Email",
+      description: "Recipient email address(es)",
+    },
+    subject: {
+      type: "string",
+      label: "Subject",
+      description: "Message subject",
+    },
+    text: {
+      type: "string",
+      label: "Text",
+      description: "Message body (plain text)",
+    },
+    html: {
+      type: "string",
+      label: "HTML",
+      description: "Message body (HTML)",
+    },
+    testMode: {
+      type: "boolean",
+      default: true,
+      description: "Enables sending in test mode",
+    },
+    dkim: {
+      type: "boolean",
+      label: "DKIM",
+      default: true,
+      description: "Enables or disables DKIM signatures",
+      optional: true,
+    },
+    tracking: {
+      type: "boolean",
+      default: true,
+      description: "Enables or disables tracking",
+      optional: true,
+    },
+    haltOnError: {
```

Comment:
As an aside, this led to a good discussion internally. I think we're likely to make this a step-level primitive (such that you can control the error-handling logic for any step native to the platform). For now this pattern is fine.